### ISSUE TITLE
Enhancement

### ISSUE THREAD

**Issue by ferrine** [ezaquarii/openvpn-at-home#10]
Got this Docker running on my local machine. 

- I was confused to be able to add only one server via main page, it was able to add only clients afterwards.
- tls key model is probably missing
- distributed (multiserver) client config seems to be not supported, but easy to fix
- user registration is too fast and he can download a config immediately (bad thing), admin approve seems to be a better choice
- no ports in configuration - bad thing, sometimes default ports are blocked, you need some always open like 80/433
- templates can be editable to make conviguration more flexible
- no revoke implemented?

I'm about to write another web app that shares VPN with friends. Several points that stop me from contributing here:

- I want a much more configurable server/client config, nothing here yet but can be implemented
- I want to give vpn upon approved request, not only close friends will use it but colleagues. I have to know to whom I issue cert
- same reason for revoking

**Comment by ezaquarii** [comment]
1. multiserver is not supported, but this is mainly UI limitation
2. registration is easy to fix - just create user with active=False and activate it in Django Admin after verification
3. server port option can be added easily to the template; server creation form must be updated with port number and template param must be added
4. you can upload your own template - just place it in templates directory; template could be sourced from db and made editable easily even via Django Admin page
5. cert revocation is not implemented and - given how OpenVPN handles it - I'm not sure how it would work

I'm generally happy to accept PRs for those features.

**Comment by ezaquarii** [comment]
Let me know if you're interested in working on those. I can provide some mentoring as most of the issues are pretty simple to resolve.

**Comment by ferrine** [comment]
Yes, I'm interested in working on this. But I'm not able to change UI, I have no js experience. I'm thinking about porting some of your core backend code and do most of the stuff in admin page, letting user only to request for a cert. Maybe the latter option is easier to implement

**Comment by ezaquarii** [comment]
I can assist you with UI changes, so no worries. Most of the features you'd like to see have pretty trivial UI or can be reasonably handled in Django Admin panel.

Let me know what you think and pls let me know which feature you consider most important to implement first.

**Comment by ferrine** [comment]
Option with my own repo looks to me more attractive. I have little experience at that scale so this can help me to get it. I really loved your repo, it gave be a lot of insights about what you can use x509 for. I hope you do not mind some copypasting with referencing you in comments. I believe that the hard way is better in my case, even without your amazing js frontend:)
